.github/workflows/system-tests.yml

```diff
@@ -28,8 +28,7 @@ jobs:
         working-directory: ./solidity
         run: |
           npx hardhat node --no-deploy &
-          HH_NODE_PID=$!
-          while ! lsof -n -Fn -p $HH_NODE_PID | grep -q '^n.*:8545$'; do sleep 10; done
+          while [[ -z $(lsof -i :8545 -t) ]]; do echo "Waiting for port 8545 to be open..."; sleep 10; done
```

Comment:
When running `npx hardhat node --no-deploy &` on node v18, two processes are spawned (a parent and a child). As a result, the previous solution for waiting for the Hardhat node to start is no longer effective. I've updated the approach: now we simply wait for port `8545` to become available.

Reply:
Makes sense!